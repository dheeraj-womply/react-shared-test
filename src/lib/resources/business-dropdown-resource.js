export const businessDropdown = [
    {
    "id" : 3996325,
    "slug" : "authentic-plaster-and-til-ccpm",
    "name" : "AUTHENTIC PLASTER AND TIL",
    "address" : {
        "address1" : "1775 NORTHFORK LANE",
        "address2" : null,
        "city" : "LUCAS",
        "state" : "TX",
        "zip" : "75002-0000"
    },
    "category" : {
        "id" : 369,
        "name" : "Pools and Pool Supply Stores"
    },
    "coordinates" : {
        "latitude" : null,
        "longitude" : null
    },
    "price" : null,
    "analytics" : {
        "is_trial" : false,
        "trial_end_date" : null,
        "claimed_at" : null,
        "sales_rep_id" : null,
        "claimed_at_month" : null
    },
    "business_phone" : "9726701028",
    "website_url" : null,
    "email_settings" : {
        "monthly_business_summaries" : false,
        "reputation_alerts" : false,
        "weekly_business_summaries" : false,
        "weekly_competitor_summaries" : false
    },
    "business_source" : "CREDIT_CARD_PROCESSOR"
},
    {
    "id" : 3583577,
    "slug" : "orleans-auto-center-ccpm",
    "name" : "ORLEANS AUTO CENTER",
    "address" : {
        "address1" : "3841 1st St",
        "address2" : null,
        "city" : "ORLEANS",
        "state" : "MA",
        "zip" : "02653"
    },
    "category" : {
        "id" : 16,
        "name" : "Car Dealers"
    },
    "coordinates" : {
        "latitude" : null,
        "longitude" : null
    },
    "price" : null,
    "analytics" : {
        "is_trial" : false,
        "trial_end_date" : null,
        "claimed_at" : "2018-04-03",
        "sales_rep_id" : "ahuerta@womply.com",
        "claimed_at_month" : "04/2018"
    },
    "business_phone" : "+15082407978",
    "website_url" : "https://www.womply.com/",
    "email_settings" : {
        "monthly_business_summaries" : true,
        "reputation_alerts" : true,
        "weekly_business_summaries" : true,
        "weekly_competitor_summaries" : true
    },
    "business_source" : "CREDIT_CARD_PROCESSOR"
},
    {
    "id" : 3581390,
    "slug" : "seayama-market-llc-ccpm",
    "name" : "SEAYAMA MARKET LLC",
    "address" : {
        "address1" : "725 GEORGESVILLE RD",
        "address2" : null,
        "city" : "COLUMBUS",
        "state" : "OH",
        "zip" : "43228"
    },
    "category" : {
        "id" : 70,
        "name" : "Supermarkets and Grocery Stores"
    },
    "coordinates" : {
        "latitude" : null,
        "longitude" : null
    },
    "price" : null,
    "analytics" : {
        "is_trial" : false,
        "trial_end_date" : null,
        "claimed_at" : "2018-04-04",
        "sales_rep_id" : "alia@womply.com",
        "claimed_at_month" : "04/2018"
    },
    "business_phone" : "6144322601",
    "website_url" : null,
    "email_settings" : {
        "monthly_business_summaries" : true,
        "reputation_alerts" : false,
        "weekly_business_summaries" : false,
        "weekly_competitor_summaries" : false
    },
    "business_source" : "CREDIT_CARD_PROCESSOR"
}
];