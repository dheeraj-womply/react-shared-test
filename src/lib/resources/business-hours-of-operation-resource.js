export const mixedDays = [
    {day: 'MONDAY', openTime: '7:00 AM', closeTime: '3:00 PM'},
    {day: 'MONDAY', openTime: '4:00 PM', closeTime: '7:00 PM'},
    {day: 'FRIDAY', openTime: '7:00 AM', closeTime: '3:00 PM'},
    {day: 'SATURDAY', openTime: '7:00 AM', closeTime: '3:00 PM'},
    {day: 'SUNDAY', openTime: null, closeTime:null}
  ];

export const closedDays = [
    {day: 'MONDAY', openTime: null, closeTime: null},
    {day: 'TUESDAY', openTime: null, closeTime: null},
    {day: 'WEDNESDAY', openTime: null, closeTime: null},
    {day: 'THURSDAY', openTime: null, closeTime: null},
    {day: 'FRIDAY', openTime: null, closeTime: null},
    {day: 'SATURDAY', openTime: null, closeTime: null},
    {day: 'SUNDAY', openTime: null, closeTime: null}
  ];