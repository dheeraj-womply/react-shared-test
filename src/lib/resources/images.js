export const RESOURCE_BASE_URL = 'https://res.cloudinary.com/womply/image/upload'

/* sideNav */
export const WOMPLY_LOGO = `${RESOURCE_BASE_URL}/react-shared-components/womply_vt6xjk.png`
export const WOMPLY_LOGO_RED = `${RESOURCE_BASE_URL}/react-shared-components/womply-logo-red_tzbmeh_zqlcno.svg`
export const CLOSE_NAV_X = `${RESOURCE_BASE_URL}/react-shared-components/close-nav_yzpooo.svg`

/* Menu Thumbnails */
export const PLUS_ICON = `${RESOURCE_BASE_URL}/react-shared-components/plus_xlskcz.svg`
export const MINUS_ICON = `${RESOURCE_BASE_URL}/react-shared-components/minus_zzbvz9.svg`

/* Connections Tile */
export const WORLDPAY_ICON = `${RESOURCE_BASE_URL}/flagship-app-ui/worldpaylogo_tt2bwr.png`
export const IACCESS_ICON = `${RESOURCE_BASE_URL}/flagship-app-ui/iaccesslogo_loqjdg.png`
export const TRANSLINK_ICON = `${RESOURCE_BASE_URL}/flagship-app-ui/translink_logo_iqk6zb.png`
export const PAYMENTSHUB_ICON = `${RESOURCE_BASE_URL}/flagship-app-ui/paymentshublogo_ki5jpz.png`
export const TSYS_ICON = `${RESOURCE_BASE_URL}/flagship-app-ui/tsyslogo_hzqi0y.png`

/* Menu Thumbnails */
export const HOME_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/home_2_liw3nc.svg`
export const CONNECTIONS_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/connection_2_xcyn0g.svg`
export const PAY_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/pay_2_nogrou.svg`
export const MONITOR_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/monitor_2_qbqhsv.svg`
export const BOOST_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/boost_2_xvyhma.svg`
export const ENGAGE_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/engage_2_vq4jz9.svg`
export const OFFERS_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/offers_pkekw9.svg`
export const EMAIL_MARKETING_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/email-marketing_rt1z36.svg`
export const STOREFRONT_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/storefront_2_h7uboh.svg`
export const BILLS_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/bills_2_wiljbi.svg`
export const SETTINGS_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/settings_2_ln5me8.svg`
export const HELP_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/help_2_pkriba.svg`
export const SIGNOUT_THUMB = `${RESOURCE_BASE_URL}/react-shared-components/navigation/sign-out_2_qbvezv.svg`

/* Text Message Preview */
export const SIGNAL_WIFI_BATTERY = `${RESOURCE_BASE_URL}/react-shared-components/vitals_TextMsgPreview.svg`
export const WOMPLY_LOGO_CIRCULAR = `${RESOURCE_BASE_URL}/react-shared-components/womply_TextMsgPreview.svg`
export const INFO_ICON = `${RESOURCE_BASE_URL}/react-shared-components/info_TextMsgPreview.svg`
