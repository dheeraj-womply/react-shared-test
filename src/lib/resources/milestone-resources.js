export const milestonesConfig = {
    "milestones" : [ {
    "key" : "SEND_GET_REVIEW_REQUEST",
    "description" : "send a get-more-reviews request",
    "displayText" : "Send a review request",
    "urlPage" : "/get_reviews",
    "urlSubDomain" : '',
    "sortIndex" : 1000,
    "productId" : 6,
    "completedTime" : 0
}, {
    "key" : "REPLY_TO_REVIEW",
    "description" : "reply to a review",
    "displayText" : "Reply to a review",
    "urlPage" : "/reviews",
    "urlSubDomain" : '',
    "sortIndex" : 1100,
    "productId" : 6,
    "completedTime" : 0
}, {
    "key" : "SAVE_VAULT_CREDENTIALS",
    "description" : "save creds in vault",
    "displayText" : "Save your credentials",
    "urlPage" : "/reputation_settings",
    "urlSubDomain" : '',
    "sortIndex" : 1200,
    "productId" : 9,
    "completedTime" : 0
}, {
    "key" : "TURN_ON_AUTO_REPLY",
    "description" : "turn on auto reply",
    "displayText" : "Turn on Auto Reply",
    "urlPage" : "/reputation_settings",
    "urlSubDomain" : '',
    "sortIndex" : 1300,
    "productId" : 6,
    "completedTime" : 0
}, {
    "key" : "SEND_CUSTOMER_MESSAGE",
    "description" : "send a customer a message from Customer Directory",
    "displayText" : "Send customer a message",
    "urlPage" : "/top-customers",
    "urlSubDomain" : '',
    "sortIndex" : 1400,
    "productId" : 14,
    "completedTime" : 0
}, {
    "key" : "TURN_CUSTOMER_ENGAGEMENT_ON",
    "description" : "turn on customer engagement",
    "displayText" : "Turn on Customer Engagement",
    "urlPage" : "/preferences",
    "urlSubDomain" : '',
    "sortIndex" : 1510,
    "productId" : 16,
    "completedTime" : 1523635987000
}, {
    "key" : "SAVE_CONTACT_CREDENTIALS",
    "description" : "save contact creds to vault",
    "displayText" : "Connect your customers",
    "urlPage" : "/simplify-credentials",
    "urlSubDomain" : '',
    "sortIndex" : 1600,
    "productId" : 9,
    "completedTime" : 0
}, {
    "key" : "UPLOAD_EMAIL_LIST",
    "description" : "bulk email upload",
    "displayText" : "Add your customer list",
    "urlPage" : "/preferences",
    "urlSubDomain" : '',
    "sortIndex" : 1700,
    "productId" : 14,
    "completedTime" : 0
}, {
    "key" : "VERIFY_EMAIL",
    "description" : "verify your email address",
    "displayText" : "Verify your email address",
    "urlPage" : "/user-profile",
    "urlSubDomain" : '',
    "sortIndex" : 1800,
    "productId" : 9,
    "completedTime" : 0
} ],
    "milestoneCount" : 9,
    "completedMilestoneCount" : 1,
    "pollingPages" : [ {
    "milestoneKey" : "VERIFY_EMAIL",
    "urlPage" : "/user-profile",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "VERIFY_EMAIL",
    "urlPage" : "/billing",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SEND_GET_REVIEW_REQUEST",
    "urlPage" : "/get_reviews",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "REPLY_TO_REVIEW",
    "urlPage" : "/reviews",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "TURN_ON_AUTO_REPLY",
    "urlPage" : "/reputation_settings",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SAVE_VAULT_CREDENTIALS",
    "urlPage" : "/reputation_settings",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "UPLOAD_EMAIL_LIST",
    "urlPage" : "/reputation_settings",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "VERIFY_EMAIL",
    "urlPage" : "/reputation_settings",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SAVE_VAULT_CREDENTIALS",
    "urlPage" : "/simplify-credentials",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "TURN_CUSTOMER_ENGAGEMENT_ON",
    "urlPage" : "/preferences",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SEND_CUSTOMER_MESSAGE",
    "urlPage" : "/top-customers",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SAVE_CONTACT_CREDENTIALS",
    "urlPage" : "/simplify-credentials",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SAVE_CONTACT_CREDENTIALS",
    "urlPage" : "/preferences",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SAVE_CONTACT_CREDENTIALS",
    "urlPage" : "/top-customers",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SEND_GET_REVIEW_REQUEST",
    "urlPage" : "/top-customers",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SEND_GET_REVIEW_REQUEST",
    "urlPage" : "/profile",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "SEND_CUSTOMER_MESSAGE",
    "urlPage" : "/profile",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "UPLOAD_EMAIL_LIST",
    "urlPage" : "/top-customers",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "VERIFY_EMAIL",
    "urlPage" : "/preferences",
    "urlSubDomain" : ''
}, {
    "milestoneKey" : "UPLOAD_EMAIL_LIST",
    "urlPage" : "/preferences",
    "urlSubDomain" : ''
} ]
};