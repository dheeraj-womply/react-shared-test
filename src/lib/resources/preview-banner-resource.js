import { DEFAULT_FONT_FAMILY, DEFAULT_FONT_FAMILY_BOLD } from '../themes/default-styles'

export const promoBanner = {
    bannerAlignment: '', 
    displaySequence: '',
    bannerType: 'PROMO',
    isEnabled: false,
    cssStyles: `{ width: 300px; height: 250px; background-color: #26BA8F; display: flex; padding: 1rem; color: #ffffff; flex-flow:column nowrap; }`, 
    bannerFields: [
        {
            bannerFieldId:"2",
            bannerType: 'PROMO', 
            bannerFieldName: 'text', 
            bannerFieldValue: 'Get 50% off', 
            cssStyles: `{ display: flex; align-items: center; text-align: center; color: #FFFFFF; font-family: ${DEFAULT_FONT_FAMILY_BOLD}; font-style: normal; font-weight: bold; font-size: 36px; margin: auto; line-height: 40px;}`, 
            clickToAction: ''
        }, 
        {
            bannerFieldId: "3",
            bannerType: 'PROMO', 
            bannerFieldName: 'text', 
            bannerFieldValue: 'on Everything', 
            cssStyles: `{ display: flex; font-family: GigV2; margin: auto; align-items: center; text-align: center; color: #FFFFFF; font-style: normal; font-weight: normal; font-size: 48px; line-height: 40px; }`, 
            clickToAction: ''
        },
        {
            bannerFieldId:"4",
            bannerType: 'PROMO', 
            bannerFieldName: 'promoCode', 
            bannerFieldValue: 'GET50OFF', 
            cssStyles: `{ background-color: transparent;  margin-top: auto; font-size: 16px;  text-align: center; color: #ffffff; padding: 10px; border: 1px solid #ffffff; border-radius: 3px;  width: 100%; font-family: ${DEFAULT_FONT_FAMILY_BOLD}; letter-spacing: 1px; word-wrap: break-word !important; outline: 0; cursor: pointer; }`, 
            clickToAction: ''
        }
    ]
};

export const alertsBanner = {
    bannerAlignment: '', 
    displaySequence: '',
    bannerType: 'ALERTS',
    isEnabled: false,
    cssStyles: `{width: 100%;min-height: 100px;background: #26BA8F;display: flex;padding: 1rem;flex-direction: column;font-size: 30px;line-height: 30px; align-items: stretch;text-align: center;@media only screen and (min-width: 768px) {    flex-direction:row;}}`, 
    bannerFields: [
        {
            bannerFieldId: "1",
            bannerType: 'ALERTS', 
            bannerFieldName: 'text', 
            bannerFieldValue: 'Now offering Delivery!', 
            cssStyles: `{ color: #ffffff;  font-size: 40px; min-width: 400px; margin: auto; display: flex; flex-grow: 5; line-height: 40px; font-family: GigV2; @media only screen and (max-width: 768px) {margin-bottom : 2rem; min-width: 0}}`, 
            clickToAction: ''
        },
        {
            bannerFieldId: "1",
            bannerType: 'ALERTS', 
            bannerFieldName: 'text', 
            bannerFieldValue: "We are excited to announce that you can now order from our menu for delivery. We can also provide orders for pickup. Give us a call or place your order online. We look forward to feeding you!", 
            cssStyles: `{ color: #ffffff;  font-size: 40px; font-size: 16px; line-height: 22px; display: flex; align-items: center; margin: auto; flex-grow: 7; font-family: ${DEFAULT_FONT_FAMILY};}`, 
            clickToAction: ''
        }
    ]
};

export const promoNewBanner = {
  bannerAlignment: '', 
  displaySequence: '',
  bannerType: 'PROMO_NEW',
  isEnabled: false,
  cssStyles: `{ width: 100%; height: 250px; background-color: #D4F1E9; display: flex; padding: 1rem; color: #000000; flex-flow:column nowrap; }`, 
  bannerFields: [
      {
          bannerFieldId:"2",
          bannerType: 'PROMO_NEW', 
          bannerFieldName: 'text', 
          bannerFieldValue: '$10 Off Any Purchase', 
          cssStyles: `{ display: flex; align-items: center; text-align: center; color: #000000; font-family: ${DEFAULT_FONT_FAMILY_BOLD}; font-style: normal; font-weight: bold; font-size: 32px; margin: auto; line-height: 40px;}`, 
          clickToAction: ''
      }, 
      {
          bannerFieldId: "3",
          bannerType: 'PROMO_NEW', 
          bannerFieldName: 'text', 
          bannerFieldValue: 'of $50 or some', 
          cssStyles: `{ display: flex; font-family: ${DEFAULT_FONT_FAMILY}; margin: auto; align-items: center; text-align: center; color: #000000; font-style: normal; font-weight: normal; font-size: 24px; line-height: 40px; }`, 
          clickToAction: ''
      },
      {
          bannerFieldId:"4",
          bannerType: 'PROMO_NEW', 
          bannerFieldName: 'promoCode', 
          bannerFieldValue: 'GET10OFF', 
          cssStyles: `{ background-color: #000000;  margin-top: auto; font-size: 16px;  text-align: center; color: #FFFFFF; padding: 10px; border: 1px solid #000000; border-radius: 3px;  width: 100%; font-family: ${DEFAULT_FONT_FAMILY_BOLD}; letter-spacing: 1px; word-wrap: break-word !important; outline: 0; cursor: pointer; max-width: 300px; margin: auto; &.active { background-color: transparent; color: #000; } }`, 
          clickToAction: ''
      }
  ]
};

export const alertsNewBanner = {
  bannerAlignment: '', 
  displaySequence: '',
  bannerType: 'ALERTS_NEW',
  isEnabled: false,
  cssStyles: `{width: 100%;min-height: 100px;display: flex;padding: 1rem;flex-direction: column;font-size: 30px;line-height: 30px; align-items: stretch;text-align: center;@media only screen and (min-width: 768px) { background: url(https://res.cloudinary.com/womply/image/upload/v1624896852/storefront/assets/icons/announcement-icon.svg) no-repeat right #D4F1E9;    background-repeat: no-repeat;background-size: 75px;padding-right: 75px; flex-direction:row;}@media only screen and (max-width: 768px) {background: url(https://res.cloudinary.com/womply/image/upload/v1624896168/storefront/assets/banners/star_dumlei.png) no-repeat #D4F1E9 15px 5px;background-size: 25px;padding-top: 35px;}}`, 
  bannerFields: [
      {
          bannerFieldId: "11",
          bannerType: 'ALERTS_NEW', 
          bannerFieldName: 'text', 
          bannerFieldValue: 'Now offering Deliveriess!', 
          cssStyles: `{ color: #000000;  font-size: 24px; margin: auto; text-align: center; flex: 3; line-height: 32px; font-family: ${DEFAULT_FONT_FAMILY_BOLD}; @media only screen and (max-width: 768px) {margin: 0 0 10px 0; min-width: 0; text-align: left}}`, 
          clickToAction: ''
      },
      {
          bannerFieldId: "12",
          bannerType: 'ALERTS_NEW', 
          bannerFieldName: 'text', 
          bannerFieldValue: "We are excited to announce that you can now order from our menu for delivery. We can also provide orders for pickup. Give us a call or place your order online. We look forward to feeding you!", 
          cssStyles: `{ color: #000000;  font-size: 40px; padding: 0 20px; font-size: 16px; line-height: 24px; display: flex; align-items: center; margin: auto; flex: 8; font-family: ${DEFAULT_FONT_FAMILY_BOLD}; @media only screen and (max-width: 768px) { margin: 0; text-align: left; padding-left: 0}}`, 
          clickToAction: ''
      }
  ]
};