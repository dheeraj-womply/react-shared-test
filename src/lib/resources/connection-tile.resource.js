import { WORLDPAY_ICON, IACCESS_ICON, TRANSLINK_ICON, PAYMENTSHUB_ICON, TSYS_ICON } from './images'
import { TileStatus } from '../components/connection-tile/interfaces'

export const tileSuccess = {
    label: 'Worldpay',
    icon: WORLDPAY_ICON,
    status: TileStatus.CONNECTED_VALID
}

export const tileError = {
    label: 'iAccess',
    icon: IACCESS_ICON,
    status: TileStatus.CONNECTED_INVALID
}

export const tilePending = {
    label: 'TransLink',
    icon: TRANSLINK_ICON,
    status: TileStatus.CONNECTION_PENDING
}

export const tileDefault1 = {
    label: 'Payments Hub',
    icon: PAYMENTSHUB_ICON,
    status: TileStatus.NOT_CONNECTED
}

export const tileDefault2 = {
    label: 'TSYS',
    icon: TSYS_ICON,
    status: TileStatus.UNABLE_TO_CONNECT
}

export const tileNoIcon = {
    label: 'Payscape',
    status: TileStatus.UNABLE_TO_CONNECT
}