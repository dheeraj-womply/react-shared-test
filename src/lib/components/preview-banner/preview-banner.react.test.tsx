import React from 'react';
import renderer from 'react-test-renderer';
import { PreviewBanner } from '.';
import { promoBanner, alertsBanner } from '../../resources/preview-banner-resource';
import { render, fireEvent } from '@testing-library/react';

test('Business Hours of operation component renders as expected with text+code banner data and scale=0.5', () => {
    const component = renderer.create(
        <PreviewBanner 
            banner={promoBanner}
            scale={0.5}
            />
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
});

test('Business Hours of operation component renders as expected with text only banner data and scale=1', () => {
    const component = renderer.create(
        <PreviewBanner 
            banner={alertsBanner}
            scale={1}
            />
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
});


test('Open popover and click cancel button', async () => {
    const { findByTestId, container, getByText } = render(
        <PreviewBanner 
            banner={promoBanner}
            scale={0.5}
            />
    );
    fireEvent.click(getByText(/Show Promo Code/i));
    expect(getByText(/GET50OFF/i)).toBeInTheDocument()
});
