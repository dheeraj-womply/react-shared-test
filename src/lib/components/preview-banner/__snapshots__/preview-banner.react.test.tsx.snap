// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business Hours of operation component renders as expected with text only banner data and scale=1 1`] = `
<div
  className="sc-AxjAm iaFobs"
  scale={1}
>
  <div
    className="sc-AxjAm heGPby"
  >
    Now offering Delivery!
  </div>
  <div
    className="sc-AxjAm gjUxoE"
  >
    We are excited to announce that you can now order from our menu for delivery. We can also provide orders for pickup. Give us a call or place your order online. We look forward to feeding you!
  </div>
</div>
`;

exports[`Business Hours of operation component renders as expected with text+code banner data and scale=0.5 1`] = `
<div
  className="sc-AxjAm dMDgJn"
  scale={0.5}
>
  <div
    className="sc-AxjAm fseFpO"
  >
    Get 50% off
  </div>
  <div
    className="sc-AxjAm iJbpfP"
  >
    on Everything
  </div>
  <div
    className="sc-AxjAm byVOPy"
    onClick={[Function]}
  >
    Show Promo Code
  </div>
</div>
`;
